common/djangoapps/student/views.py

```diff
@@ -2285,6 +2291,39 @@ def auto_auth(request):
 @ensure_csrf_cookie
 def activate_account(request, key):
     """When link in activation e-mail is clicked"""
+
+    # If request is in Studio call the appropriate view
+    if theming_helpers.get_project_root_name().lower() == u'cms':
+        return activate_account_studio(request, key)
+
+    regs = Registration.objects.filter(activation_key=key)
+    if len(regs) == 1:
+        if not regs[0].user.is_active:
+            regs[0].activate()
+            # Add account activation success message for display later
+            messages.success(
+                request, 'You successfully activated your account.', extra_tags='account-activation',
```

Comment:
I used the same text from the snapshot on the ticket [here](https://openedx.atlassian.net/browse/ENT-344).